### ISSUE TITLE
feat: Image-to-Image LoRA training

### ISSUE THREAD

**Issue by VincentGourbin** [VincentGourbin/flux-2-swift-mlx#51]
## Problem

Currently, LoRA training only supports **Text-to-Image (T2I)**: given images and captions, the model learns to generate images from text alone. There is no support for training LoRAs in **Image-to-Image (I2I)** mode, where the model learns to **transform** a source image into a target image conditioned by a text instruction.

## Use Cases

- **Image editing**: "remove the hat", "change background to beach"
- **Style transfer guided by reference**: transform photos using a specific style
- **Inpainting**: fill in regions guided by surrounding context
- **Domain transfer**: convert sketches to photos, day to night, etc.

## Technical Approach

Flux.2 handles I2I via **sequence concatenation** (not noise injection like Stable Diffusion):

1. The reference image is encoded by the VAE as **clean tokens** (no noise)
2. These tokens are **concatenated** along the sequence dimension with the output (noisy) tokens
3. Each reference image gets a unique **T-coordinate** in the 4D RoPE position encoding (T=10, T=20, etc.)
4. The output image has T=0, text has L=0,1,2...
5. The transformer processes all tokens together via attention
6. After the forward pass, only the **output portion** of the noise prediction is extracted for the loss

This mechanism already exists in inference (`Flux2Pipeline.generateImageToImage`). This feature transposes it to the training loop.

### Configuration
- `control_path`: directory containing control/source images (paired with targets by filename)
- `control_dropout`: probability of dropping control (for T2I robustness, default 0.0)
- `reference_image`: per-validation-prompt reference for I2I validation

### Data Pipeline
- Pre-encode control images via VAE (same Ostris normalization as targets)
- Store in separate `.control_latent_cache/` directory
- Attach `controlLatent` to each `CachedLatentEntry`

### Training Loop
- Concatenate packed control latents (clean) with packed target latents (noisy) along sequence dimension
- Generate reference image position IDs with `generateSingleReferenceImagePositionIDs(T=10)`
- Extract output portion from model prediction before computing loss
- DOP path also handles concatenated sequences
- Support control dropout for T2I robustness

### Backward Compatibility
- `controlPath: nil` = pure T2I mode (no behavior change)
- All existing YAML configs work without modification

## Memory Impact

Each I2I step processes ~2x longer sequences (output + control tokens). For 512x512:
- T2I: 256 tokens → I2I: 512 tokens
- Attention scales O(n²), so ~4x more attention memory
- Klein 4B recommended for I2I (peak ~50-55GB vs ~42GB for T2I)
- Klein 9B may OOM on 96GB in I2I mode

## Reference

ai-toolkit (Ostris) supports this mode via paired `targets/` + `controls/` directories with filename matching.

## Example YAML

```yaml
dataset:
  path: datasets/my-editing/targets
  control_path: datasets/my-editing/controls
  control_dropout: 0.3

validation:
  prompts:
    - prompt: "EDITCMD remove the background"
      reference_image: datasets/my-editing/controls/test_001.png
      is_512: true
```
